Mergin/switch_sources_dialog.py

```diff
@@ -0,0 +1,202 @@
+
+            if layer_name in self.db_tables:
+                uri = QgsDataSourceUri(self.db_connection_info)
+                uri.setSchema(self.db_schema)
+                uri.setTable(layer_name)
+                uri.setGeometryColumn("geom")  # TODO should this be hardcoded?
+                gpkg_layer.setDataSource(uri.uri(), gpkg_layer.name(), "postgres")
+
+    def convert_to_gpkg_layer(self, postgresql_layer: QgsVectorLayer, gpkg_folder: str) -> None:
+        gpkg = QgsVectorLayer(f"{gpkg_folder}/{self.sync_file}", "temp", "ogr")
+        gpkg_layers = [x.split("!!::!!")[1] for x in gpkg.dataProvider().subLayers()]
+
+        table_name = postgresql_layer.dataProvider().uri().table()
+
+        if table_name in gpkg_layers:
+            uri = f"{gpkg_folder}/{self.sync_file}|layername={table_name}"
+
+            postgresql_layer.setDataSource(uri, postgresql_layer.name(), "ogr")
+
+
+class DBSyncConfigSelectionPage(ui_select_dbsync_page, base_select_dbsync_page):
+    """Initial wizard page with selection od dbsync file selector."""
+
+    selectDbSyncConfig: QgsFileWidget
+    ldbsync_config_file: QLineEdit
+
+    def __init__(self, parent=None):
+        super().__init__(parent)
+        self.setupUi(self)
+        self.parent = parent
+
+        self.ldbsync_config_file.hide()
+
+        self.registerField("db_sync_file*", self.ldbsync_config_file)
+
+        self.selectDbSyncConfig.setFilter("DBSync configuration files (*.yaml *.YAML)")
+        self.selectDbSyncConfig.fileChanged.connect(self.db_sync_config)
+
+    def db_sync_config(self, path: str) -> None:
+        self.ldbsync_config_file.setText(path)
+
+    def nextId(self):
+        return QGS_PROJECT_PAGE
+
+
+class QgsProjectSelectionPage(ui_select_qgsproject_page, base_select_qgsproject_page):
+    """Wizard page with selection od QgsProject file selector."""
+
+    selectQgisProject: QgsFileWidget
+    lqgsproject_file: QLineEdit
+
+    def __init__(self, parent=None):
+        super().__init__(parent)
+        self.setupUi(self)
+        self.parent = parent
+
+        self.lqgsproject_file.hide()
+
+        self.registerField("qgis_project*", self.lqgsproject_file)
+
+        self.selectQgisProject.setFilter("QGIS files (*.qgz *.qgs *.QGZ *.QGS)")
+        self.selectQgisProject.setStorageMode(QgsFileWidget.StorageMode.SaveFile)
+        self.selectQgisProject.fileChanged.connect(self.qgis_project)
+
+    def qgis_project(self, path: str) -> None:
+        self.lqgsproject_file.setText(path)
+
+
+class ProjectUsePostgreConfigWizard(QWizard):
+    """Wizard for changing project layer sources from GPKG to Postgre DB."""
+
+    def __init__(self, iface: QgisInterface, parent=None):
+        """Create a wizard"""
+        super().__init__(parent)
+
+        self.iface = iface
+        self.setWindowTitle("Create project with layers from Postgre Database")
+
+        self.connections: typing.List[Connection] = []
+
+        self.qgis_project = QgsProject.instance()
```

Comment:
let's not store reference to QgsProject.instance() to self.qgis_project - simply use QgsProject.instance() where needed, it will be easier to read the code